src/action/finder/request.rs

```diff
@@ -0,0 +1,1405 @@
+            max_size: proto.max_size(),
+            collect_ext_attrs: proto.collect_ext_attrs(),
+            use_external_stores: proto.use_external_stores(),
+            chunk_size: proto.chunk_size(),
+        }))
+    }
+}
+
+impl ProtoEnum<PathType> for PathType {
+    fn default() -> PathType {
+        FileFinderArgs::default().pathtype()
+    }
+    fn from_i32(val: i32) -> Option<PathType> {
+        PathType::from_i32(val)
+    }
+}
+
+impl ProtoEnum<ActionType> for ActionType {
+    fn default() -> Self {
+        FileFinderAction::default().action_type()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        ActionType::from_i32(val)
+    }
+}
+
+impl ProtoEnum<HashActionOversizedFilePolicy>
+    for HashActionOversizedFilePolicy
+{
+    fn default() -> Self {
+        FileFinderHashActionOptions::default().oversized_file_policy()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        HashActionOversizedFilePolicy::from_i32(val)
+    }
+}
+
+impl ProtoEnum<DownloadActionOversizedFilePolicy>
+    for DownloadActionOversizedFilePolicy
+{
+    fn default() -> Self {
+        FileFinderDownloadActionOptions::default().oversized_file_policy()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        DownloadActionOversizedFilePolicy::from_i32(val)
+    }
+}
+
+impl ProtoEnum<XDevMode> for XDevMode {
+    fn default() -> Self {
+        FileFinderArgs::default().xdev()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        XDevMode::from_i32(val)
+    }
+}
+
+impl ProtoEnum<ConditionType> for ConditionType {
+    fn default() -> Self {
+        FileFinderCondition::default().condition_type()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        ConditionType::from_i32(val)
+    }
+}
+
+impl ProtoEnum<RegexMatchMode> for RegexMatchMode {
+    fn default() -> Self {
+        FileFinderContentsRegexMatchCondition::default().mode()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        RegexMatchMode::from_i32(val)
+    }
+}
+
+impl From<RegexMatchMode> for MatchMode {
+    fn from(proto: RegexMatchMode) -> Self {
+        match proto {
+            RegexMatchMode::FirstHit => MatchMode::FirstHit,
+            RegexMatchMode::AllHits => MatchMode::AllHits,
+        }
+    }
+}
+
+impl ProtoEnum<LiteralMatchMode> for LiteralMatchMode {
+    fn default() -> Self {
+        FileFinderContentsLiteralMatchCondition::default().mode()
+    }
+    fn from_i32(val: i32) -> Option<Self> {
+        LiteralMatchMode::from_i32(val)
+    }
+}
+
+impl From<LiteralMatchMode> for MatchMode {
+    fn from(proto: LiteralMatchMode) -> Self {
+        match proto {
+            LiteralMatchMode::FirstHit => MatchMode::FirstHit,
+            LiteralMatchMode::AllHits => MatchMode::AllHits,
+        }
+    }
+}
+
+fn get_modification_time_conditions(
+    proto: Option<FileFinderModificationTimeCondition>,
+) -> Result<Vec<Condition>, ParseError> {
+    Ok(match proto {
+        Some(options) => {
```

Comment:
Nesting in these functions is a bit too ridiculous, try to make it less deep (e.g. by using early returns).

Reply:
Reduced it by using 2 "if let" expressions